tests/utxo_behavior.rs

```diff
@@ -0,0 +1,190 @@
+#![cfg(feature = "integration-test")]
+use bitcoin::Amount;
+use coinswap::{
+    maker::MakerBehavior,
+    taker::TakerBehavior,
+    utill::{ConnectionType, MIN_FEE_RATE},
+};
+mod test_framework;
+use std::sync::atomic::Ordering::Relaxed;
+use test_framework::*;
+
+// Test data: Create addresses with different UTXO patterns
+const UTXO_SETUP: &[(u8, &[f64])] = &[
+    (1, &[0.2, 0.2, 0.2]),
+    (2, &[0.5, 0.2]),
+    (3, &[0.1]),
+    (4, &[0.3]),
+    (5, &[0.5]),
+];
+
+const TEST_CASES: &[(f64, f64, &str, &str)] = &[
+    (
+        0.4,
+        0.6,
+        "Case 1: Low target",
+        "Should select 0.6 BTC group (smallest that covers target+fees)",
+    ),
```

Comment:
Expected utxo list needs to be added here. 

```suggestion
    (
        0.4,
        0.6,
        [0.2, 0.2, 0.2], // expected utxos
        "Case 1: Low target",
        "Should select 0.6 BTC group (smallest that covers target+fees)",
    ),
```

And then match them with the vec of values we getting from the coin selection. 

Add these assertions first, its okay if it fails. Then keep changing the internal behavior until it passes.